platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationImpl.java

```diff
@@ -63,48 +63,57 @@
     public ApplicationImpl(Repository repo) {
         location = repo.getURI();
         try {
+            String[] parts = repo.getName().split("-[0-9]");
+            if (parts.length != 0){
+                name = parts[0];
+                version = repo.getName().substring(name.length()+1);
+            } else {
+                name = repo.getName();
+                version = "0.0.0";
+            }
             features = new HashSet<>(Arrays.asList(repo.getFeatures()));
         } catch (Exception e) {
             logger.warn(
                     "Error occured while trying to parse information for application. Application created but may have missing information.");
             features = new HashSet<>();
         }
-        List<Feature> autoFeatures = new ArrayList<>();
+        autoInstallFeatures = new HashSet<>();
         if (features.size() == 1) {
-            autoFeatures.add(features.iterator()
+            autoInstallFeatures.add(features.iterator()
                     .next());
         } else {
-            for (Feature curFeature : features) {
-                if (StringUtils.equalsIgnoreCase(Feature.DEFAULT_INSTALL_MODE,
-                        curFeature.getInstall())) {
-                    autoFeatures.add(curFeature);
-                }
-            }
+            autoInstallFeatures.addAll(features.stream()
+                    .filter(curFeature -> StringUtils.equalsIgnoreCase(Feature.DEFAULT_INSTALL_MODE,
+                            curFeature.getInstall()))
+                    .collect(Collectors.toList()));
         }
-        if (autoFeatures.size() == 1) {
-            mainFeature = autoFeatures.get(0);
+        // Determine mainFeature
+        if (autoInstallFeatures.size() == 1) {
+            mainFeature = autoInstallFeatures.iterator()
+                    .next();
             name = mainFeature.getName();
             version = mainFeature.getVersion();
             description = mainFeature.getDescription();
         } else {
-            try {
-                if (repo.getName() == null) {
-                    logger.warn(
-                            "No information available inside the repository, cannot create application instance.");
-                    throw new IllegalArgumentException(
-                            "No identifying information available inside the repository, cannot create application instance.");
+            for (Feature feature : autoInstallFeatures) {
```

Comment:
A stream/filter might be cleaner here to get the one and only one feature you want and then set the values if it's not null. Might not be cleaner; I'm not sure.